### PR TITLE
Resolves #11

### LINKED ISSUE #11: Top 5 Merchants By Revenue
As an admin,
When I visit the admin merchants index
Then I see the names of the top 5 merchants by total revenue generated
And I see that each merchant name links to the admin merchant show page for that merchant
And I see the total revenue generated next to each merchant name

Notes on Revenue Calculation:
- Only invoices with at least one successful transaction should count towards revenue
- Revenue for an invoice should be calculated as the sum of the revenue of all invoice items
- Revenue for an invoice item should be calculated as the invoice item unit price multiplied by the quantity (do not use the item unit price)

### PR DESCRIPTION
**Pull request by WadeNaughton**
-top 5 Merchants show
-revenue of each merchant showing
-update for admin merchant made restful
-tests passing for story

### CODE REVIEW

**Review by echon006 — COMMENTED** [review]
Conditional looks fire!

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -28,24 +28,17 @@ def edit
...
-      redirect_to admin_merchant_path(merchant)
-    end
-  end
-
-  def status_change
-    merchant = Merchant.find(params[:merchant_id])
-    if merchant.status == "Disabled"
-      merchant.status = 'Enabled'
-      merchant.save
+    if params[:status] == 'Disabled'
```
Comment (tjroeder) [inline app/controllers/admin/merchants_controller.rb:31]: Good update. This makes it more RESTful.

File: app/models/merchant.rb [hunk 2]
```diff
@@ -47,4 +47,17 @@ def top_five_items
     .order(revenue: :desc)
     .limit(5)
   end
+
+  def self.top_five_merchants
```
Comment (tjroeder) [inline app/models/merchant.rb:51]: Looks good.

File: spec/factories/transactions.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
   factory :transaction do
     sequence(:credit_card_number) { |n| n.to_s.rjust(16, '0') }
     sequence(:credit_card_expiration_date) { |n| (rand(n)+1).years.from_now }
-    traits_for_enum(:status, ['', 'failed', 'success'])
+    traits_for_enum(:result, ['', 'failed', 'success'])
```
Comment (tjroeder) [inline spec/factories/transactions.rb:5]: Thanks for fixing my factory!

File: spec/features/admin/merchants/index.html.erb_spec.rb [hunk 4]
```diff
@@ -35,4 +51,30 @@
         expect('Disabled Merchants').to appear_before(merch_1.name)
       end
     end
+
+    describe 'top 5 merchants by revenue' do
```
Comment (tjroeder) [inline spec/features/admin/merchants/index.html.erb_spec.rb:55]: Good test!

File: spec/features/admin/merchants/index.html.erb_spec.rb [hunk 5]
```diff
@@ -35,4 +51,30 @@
...
+        item3 = create(:item, merchant: merchants[2])
+        item4 = create(:item, merchant: merchants[3])
+        item5 = create(:item, merchant: merchants[4])
+        transactions = create_list(:transaction, 5, result: :success)
+        invoice_item1 = create(:invoice_item, item: item1, invoice: transactions[0].invoice, unit_price: 3, quantity: 2)
+        invoice_item2 = create(:invoice_item, item: item2, invoice: transactions[1].invoice, unit_price: 4, quantity: 2)
+        invoice_item3 = create(:invoice_item, item: item3, invoice: transactions[2].invoice, unit_price: 5, quantity: 2)
+        invoice_item4 = create(:invoice_item, item: item4, invoice: transactions[3].invoice, unit_price: 6, quantity: 2)
+        invoice_item5 = create(:invoice_item, item: item5, invoice: transactions[4].invoice, unit_price: 1, quantity: 2)
+        blahblah = [merchants[3], merchants[2], merchants[1], merchants[0], merchants[4]]
```
Comment (tjroeder) [inline spec/features/admin/merchants/index.html.erb_spec.rb:69]: Haha.

**Review by tjroeder — APPROVED** [review]
Changes look good.

**Pull request merged**